docs/spec/spec.md

```diff
@@ -452,3 +385,44 @@ status:
 
   observedGeneration: ...  # last generation being reconciled
 ```
+
+## Container
+
+This is a [core.v1.Container](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.10/#container-v1-core).
+Some fields are not allowed, such as name and volumeMounts.
+
+This type is not used on its own but is found composed inside [Service](#service), [Configuration](#configuration), and [Revision](#revision).
+
+```yaml
+container: # v1.Container
+  # image either provided as pre-built container, or built by Knative Serving from
+  # source. When built by knative, set to the same as build template, e.g.
+  # build.template.arguments[_IMAGE], as the "promise" of a future build.
+  # If buildRef is provided, it is expected that this image will be
+  # present when the referenced build is complete.
+  image: gcr.io/...
+  command: ['run']
+  args: []
+  env:
+  # list of environment vars
+  - name: FOO
+    value: bar
+  - name: HELLO
+    value: world
+  - ...
+
+  # Optional, only a single containerPort may be specified.
+  # This can be used if your container cannot use the
```

Comment:
This sentence is a bit unclear as written as the $PORT variable is set regardless of whether it is the default port or customized here. Maybe something along the lines of:

"This can be specified to select a specific port for incoming traffic. This is useful if your application cannot discover the port to listen on through the $PORT environment variable that is always set within the container."